ui/components/app/wallet-overview/coin-overview.tsx

```diff
@@ -142,25 +218,94 @@ export const CoinOverview = ({
-                  className={classnames(
-                    `${classPrefix}-overview__primary-balance`,
-                    {
-                      [`${classPrefix}-overview__cached-balance`]:
-                        balanceIsCached,
-                    },
-                  )}
-                  data-testid={`${classPrefix}-overview__primary-currency`}
-                  value={balanceToDisplay}
-                  type={PRIMARY}
-                  ethNumberOfDecimals={4}
-                  hideTitle
-                  shouldCheckShowNativeToken
-                  isAggregatedFiatOverviewBalance={
-                    !showNativeTokenAsMainBalance
-                  }
-                />
+                <>
+                  <Box onMouseEnter={handleMouseEnter} ref={setBoxRef}>
+                    <UserPreferencedCurrencyDisplay
+                      style={{ display: 'contents' }}
+                      className={classnames(
+                        `${classPrefix}-overview__primary-balance`,
+                        {
+                          [`${classPrefix}-overview__cached-balance`]:
+                            balanceIsCached,
+                        },
+                      )}
+                      data-testid={`${classPrefix}-overview__primary-currency`}
+                      value={balanceToDisplay}
+                      type={PRIMARY}
+                      ethNumberOfDecimals={4}
+                      hideTitle
+                      shouldCheckShowNativeToken
+                      isAggregatedFiatOverviewBalance={
+                        !showNativeTokenAsMainBalance && !isTestnet
+                      }
+                    />
+                  </Box>
+                  {shouldShowPopover &&
+                  !isTestnet &&
+                  !showNativeTokenAsMainBalance ? (
+                    <Popover
+                      referenceElement={referenceElement}
+                      isOpen={isOpen}
+                      position={PopoverPosition.BottomStart} // TODO check with design-team about this bottom start issue
+                      hasArrow
+                      flip
+                      data-theme={theme === 'light' ? 'dark' : 'light'}
+                      className="balance-popover__container"
+                      padding={3}
+                      onClickOutside={handleClick}
+                      onPressEscKey={handleClick}
+                      preventOverflow
+                    >
+                      <Box>
+                        <Box
+                          display={Display.Flex}
+                          justifyContent={JustifyContent.spaceBetween}
+                        >
+                          <Text
+                            variant={TextVariant.bodySmBold}
+                            textAlign={TextAlign.Left}
+                            alignItems={AlignItems.flexStart}
+                            className="balance-popover__text"
```

Comment:
Can we follow responsive typography here and avoid locking the font-size to 12px? It should scale up on larger screens but stay at 12px on smaller screens.

https://github.com/user-attachments/assets/df120ff5-06a5-430b-ba8e-29eecee8eab4

Reply:
Thanks @georgewrmarshall ! fixed here [5e720c6](https://github.com/MetaMask/metamask-extension/pull/27161/commits/5e720c6ca6d693fd7006e113861c9c3e0a054ade)
I removed ` className="balance-popover__text"`